<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>340</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>360</width>
    <height>270</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowIcon">
   <iconset resource="icon.qrc">
    <normaloff>:/icons/MiniPunctor.png</normaloff>:/icons/MiniPunctor.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="labelBanner">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>221</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Puncture Everything You Like!</string>
    </property>
    <property name="textFormat">
     <enum>Qt::PlainText</enum>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="labelTime">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string>Time Display</string>
    </property>
    <property name="textFormat">
     <enum>Qt::PlainText</enum>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonStart">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>186</y>
      <width>101</width>
      <height>35</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::TabFocus</enum>
    </property>
    <property name="toolTip">
     <string>Start / pause timer</string>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonStop">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>186</y>
      <width>101</width>
      <height>35</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::TabFocus</enum>
    </property>
    <property name="toolTip">
     <string>Stop timer</string>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonPunc">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>211</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="toolTip">
     <string>Insert current time tick to list</string>
    </property>
    <property name="text">
     <string>Puncture !</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listTimeline">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>30</y>
      <width>261</width>
      <height>271</height>
     </rect>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::CustomContextMenu</enum>
    </property>
    <property name="selectionMode">
     <enum>QAbstractItemView::ExtendedSelection</enum>
    </property>
   </widget>
   <widget class="QFrame" name="frameTimeAdj">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>90</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="buttonUpDown" name="bUDMin" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>61</width>
       <height>41</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Adjust minute</string>
     </property>
    </widget>
    <widget class="buttonUpDown" name="bUDSec" native="true">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>0</y>
       <width>61</width>
       <height>41</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Adjust second</string>
     </property>
    </widget>
    <widget class="buttonUpDown" name="bUDMsec" native="true">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>0</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Adjust millisecond</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frameListOperation">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>0</y>
      <width>261</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="lineWidth">
     <number>0</number>
    </property>
    <widget class="QFrame" name="frameListButtonLeft">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <widget class="QPushButton" name="buttonListEdit">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>0</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Edit selected item</string>
      </property>
      <property name="icon">
       <iconset resource="icon.qrc">
        <normaloff>:/icons/list-edit.png</normaloff>:/icons/list-edit.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>25</width>
        <height>25</height>
       </size>
      </property>
     </widget>
     <widget class="QPushButton" name="buttonListMoveDown">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>0</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Move selected item down</string>
      </property>
      <property name="icon">
       <iconset resource="icon.qrc">
        <normaloff>:/icons/go-down.png</normaloff>:/icons/go-down.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>20</width>
        <height>20</height>
       </size>
      </property>
     </widget>
     <widget class="QPushButton" name="buttonListMoveUp">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>0</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Move selected item up</string>
      </property>
      <property name="icon">
       <iconset resource="icon.qrc">
        <normaloff>:/icons/go-up.png</normaloff>:/icons/go-up.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>20</width>
        <height>20</height>
       </size>
      </property>
     </widget>
     <widget class="QPushButton" name="buttonFollowTimer">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="toolTip">
       <string>Hightlight current tick according to the timer</string>
      </property>
      <property name="icon">
       <iconset resource="icon.qrc">
        <normaloff>:/icons/timer.png</normaloff>:/icons/timer.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>25</width>
        <height>25</height>
       </size>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="frameListButtonRight">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>0</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <widget class="QPushButton" name="buttonListRemove">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>0</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Remove selected item</string>
      </property>
      <property name="icon">
       <iconset resource="icon.qrc">
        <normaloff>:/icons/list-remove.png</normaloff>:/icons/list-remove.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>20</width>
        <height>20</height>
       </size>
      </property>
     </widget>
     <widget class="QPushButton" name="buttonListInsert">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>0</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Create new item after selected item</string>
      </property>
      <property name="icon">
       <iconset resource="icon.qrc">
        <normaloff>:/icons/list-add.png</normaloff>:/icons/list-add.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>20</width>
        <height>20</height>
       </size>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QFrame" name="framePlayerCtrl">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>139</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="buttonPlayerPlay">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>41</width>
       <height>35</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::TabFocus</enum>
     </property>
     <property name="toolTip">
      <string>Play / pause music</string>
     </property>
     <property name="icon">
      <iconset resource="icon.qrc">
       <normaloff>:/icons/media-play.png</normaloff>:/icons/media-play.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonPlayerNext">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>0</y>
       <width>41</width>
       <height>35</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::TabFocus</enum>
     </property>
     <property name="toolTip">
      <string>Play next song</string>
     </property>
     <property name="icon">
      <iconset resource="icon.qrc">
       <normaloff>:/icons/media-next.png</normaloff>:/icons/media-next.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonPlayerPrevious">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>0</y>
       <width>41</width>
       <height>35</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::TabFocus</enum>
     </property>
     <property name="toolTip">
      <string>Play last song</string>
     </property>
     <property name="icon">
      <iconset resource="icon.qrc">
       <normaloff>:/icons/media-previous.png</normaloff>:/icons/media-previous.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonPlayerStop">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>0</y>
       <width>41</width>
       <height>35</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::TabFocus</enum>
     </property>
     <property name="toolTip">
      <string>Stop player</string>
     </property>
     <property name="icon">
      <iconset resource="icon.qrc">
       <normaloff>:/icons/media-stop.png</normaloff>:/icons/media-stop.png</iconset>
     </property>
    </widget>
   </widget>
   <zorder>framePlayerCtrl</zorder>
   <zorder>labelBanner</zorder>
   <zorder>labelTime</zorder>
   <zorder>buttonStart</zorder>
   <zorder>buttonStop</zorder>
   <zorder>buttonPunc</zorder>
   <zorder>listTimeline</zorder>
   <zorder>frameTimeAdj</zorder>
   <zorder>frameListOperation</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>500</width>
     <height>36</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew_File"/>
    <addaction name="separator"/>
    <addaction name="actionOpen_File"/>
    <addaction name="actionSave_File"/>
    <addaction name="actionSave_File_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionCut"/>
    <addaction name="actionPaste"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_all"/>
    <addaction name="actionReverse_selection"/>
    <addaction name="separator"/>
    <addaction name="actionFind_Replace"/>
    <addaction name="separator"/>
    <addaction name="actionInsert_before"/>
    <addaction name="actionCreate_tick_behind"/>
    <addaction name="actionDuplicate"/>
    <addaction name="separator"/>
    <addaction name="actionOptions"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuTimeline">
    <property name="title">
     <string>&amp;Timeline</string>
    </property>
    <addaction name="actionMove_up"/>
    <addaction name="actionMove_down"/>
    <addaction name="separator"/>
    <addaction name="actionSort_by_start_time"/>
    <addaction name="actionSort_by_end_time"/>
    <addaction name="actionSort_by_tick_ID"/>
    <addaction name="separator"/>
    <addaction name="actionAdjust_selected"/>
    <addaction name="actionAdjust_all_ticks"/>
    <addaction name="separator"/>
    <addaction name="actionFollow_timer"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuPlayer">
    <property name="title">
     <string>&amp;Player</string>
    </property>
    <widget class="QMenu" name="menuBind_a_player">
     <property name="title">
      <string>Bind a player</string>
     </property>
     <addaction name="action_Not_available"/>
    </widget>
    <addaction name="separator"/>
    <addaction name="menuBind_a_player"/>
    <addaction name="separator"/>
    <addaction name="actionPlay_Pause"/>
    <addaction name="actionStop"/>
    <addaction name="actionPrevious"/>
    <addaction name="actionNext"/>
    <addaction name="separator"/>
    <addaction name="actionShift_synchronization"/>
    <addaction name="actionSynchronize_with_player"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTimeline"/>
   <addaction name="menuPlayer"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionNew_File">
   <property name="text">
    <string>&amp;New File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen_File">
   <property name="text">
    <string>&amp;Open File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_File">
   <property name="text">
    <string>&amp;Save File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_File_As">
   <property name="text">
    <string>Save File &amp;As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>&amp;Options</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="about.png">
     <normalon>:/icons/help.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>&amp;Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>C&amp;ut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionDuplicate">
   <property name="text">
    <string>&amp;Duplicate tick</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionInsert_before">
   <property name="text">
    <string>Create tick &amp;before</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Space</string>
   </property>
  </action>
  <action name="actionCreate_tick_behind">
   <property name="text">
    <string>C&amp;reate tick behind</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Space</string>
   </property>
  </action>
  <action name="actionSelect_all">
   <property name="text">
    <string>Select all</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionReverse_selection">
   <property name="text">
    <string>Reverse selection</string>
   </property>
  </action>
  <action name="actionFind_Replace">
   <property name="text">
    <string>Find / Replace</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionSort_by_start_time">
   <property name="text">
    <string>Sort by s&amp;tart time</string>
   </property>
  </action>
  <action name="actionSort_by_end_time">
   <property name="text">
    <string>Sort by &amp;end time</string>
   </property>
  </action>
  <action name="actionSort_by_tick_ID">
   <property name="text">
    <string>Sort by tick &amp;ID</string>
   </property>
  </action>
  <action name="actionAdjust_selected">
   <property name="text">
    <string>Adjust &amp;selected ticks</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+E</string>
   </property>
  </action>
  <action name="actionAdjust_all_ticks">
   <property name="text">
    <string>Adjust &amp;all ticks</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionMove_up">
   <property name="text">
    <string>Move &amp;up</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Up</string>
   </property>
  </action>
  <action name="actionMove_down">
   <property name="text">
    <string>Move &amp;down</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Down</string>
   </property>
  </action>
  <action name="actionClear_all_ticks">
   <property name="text">
    <string>&amp;Clear all ticks</string>
   </property>
  </action>
  <action name="actionFollow_timer">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Follow timer</string>
   </property>
  </action>
  <action name="actionSynchronize_with_player">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Synchronize with player</string>
   </property>
  </action>
  <action name="actionShift_synchronization">
   <property name="text">
    <string>Shif&amp;t synchronization</string>
   </property>
  </action>
  <action name="actionPlay_Pause">
   <property name="text">
    <string>&amp;Play / Pause</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="text">
    <string>&amp;Stop</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="actionPrevious">
   <property name="text">
    <string>P&amp;revious</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionNext">
   <property name="text">
    <string>&amp;Next</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
  <action name="action_Not_available">
   <property name="text">
    <string>(Not available)</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>buttonUpDown</class>
   <extends>QWidget</extends>
   <header>buttonupdown.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icon.qrc"/>
 </resources>
 <connections/>
</ui>
